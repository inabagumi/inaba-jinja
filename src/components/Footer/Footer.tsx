import Link from 'next/link'
import React from 'react'
import type { FC } from 'react'
import styled from 'styled-components'

const Container = styled.footer`
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  font-size: 0.85rem;
  justify-content: space-between;
  padding: 2rem 1rem 1.5rem;

  @media (min-width: 960px) {
    flex-direction: row;
    padding-bottom: 0.5rem;
  }
`

const NavLinks = styled.ul`
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
`

const NavLinksItem = styled.li`
  :not(:first-child) {
    margin-left: 0.5em;
  }
`

const NavLink = styled.a`
  color: inherit;
  text-decoration: none;

  :hover {
    color: inherit;
    text-decoration: underline;
  }
`

const Footer: FC = () => {
  return (
    <Container>
      <nav>
        <NavLinks>
          <NavLinksItem>
            <Link href="/about" passHref prefetch={false}>
              <NavLink>因幡神社とは</NavLink>
            </Link>
          </NavLinksItem>
          <NavLinksItem>
            <Link href="/privacy" passHref prefetch={false}>
              <NavLink>プライバシーポリシー</NavLink>
            </Link>
          </NavLinksItem>
          <NavLinksItem>
            <NavLink
              className="footer__link"
              href="https://haneru.dev/"
              rel="noopener noreferrer"
              target="_blank"
            >
              運営者情報
            </NavLink>
          </NavLinksItem>
        </NavLinks>
      </nav>
    </Container>
  )
}

export default Footer
