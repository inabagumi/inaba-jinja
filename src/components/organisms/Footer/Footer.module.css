.copyright {
  margin: 0;
}

.navigation li:not(:first-child) {
  margin-left: 0.5em;
}

.navigation ul {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.root {
  align-items: flex-start;
  color: #eee;
  display: flex;
  flex-direction: column;
  font-size: 0.85rem;
  justify-content: space-between;
  padding: 2rem 1rem 1.5rem;
}

@media (min-width: 960px) {
  .root {
    flex-direction: row;
    padding-bottom: 0.5rem;
  }
}
