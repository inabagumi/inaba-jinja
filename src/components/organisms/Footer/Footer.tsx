import styled from '@emotion/styled'
import { Link } from 'gatsby'
import React, { FC } from 'react'

const Copyright = styled('p')`
  margin: 0;

  a {
    color: inherit;
    text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }
`

const Navigation = styled('nav')`
  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }
`

const Root = styled('footer')`
  align-items: flex-start;
  color: #eee;
  display: flex;
  flex-direction: column;
  font-size: 0.85rem;
  justify-content: space-between;
  padding: 2rem 1rem 0.5rem;

  @media (min-width: 960px) {
    & {
      flex-direction: row;
    }
  }
`

const Footer: FC = () => {
  return (
    <Root>
      <Copyright>
        &copy; 2020{' '}
        <a href="https://haneru.dev/" rel="noopener noreferrer" target="_blank">
          Haneru Developers
        </a>
      </Copyright>

      <Navigation>
        <ul>
          <li>
            <Link to="/privacy">プライバシーポリシー</Link>
          </li>
        </ul>
      </Navigation>
    </Root>
  )
}

export default Footer
