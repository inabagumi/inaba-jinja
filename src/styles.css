:root {
  --ij-default-font-family: -apple-system, BlinkMacSystemFont,
    Helvetica Neue, Helvetica, Arial, YuGothic, Yu Gothic, sans-serif;
  --ij-serif-font-family: Garamond, Times New Roman, YuMincho, Yu Mincho,
    serif;
}

html {
  box-sizing: border-box;
  font-family: var(--ij-default-font-family);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: none;
}

body {
  background-color: #424242;
  color: #fafafa;
  margin: 0;
  line-height: 2;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

body,
html,
#__next {
  height: 100%;
}

p {
  margin: 0;
  padding: 0;
}

a {
  color: #f57f17;
}

a:hover {
  color: #ff6f00;
}
