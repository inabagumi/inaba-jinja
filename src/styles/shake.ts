import { keyframes } from '@emotion/react'

const shake = keyframes`
  from {
    transform: translateY(0) rotate(180deg);
  }

  to {
    transform: translateY(20px) rotate(170deg);
  }
`

export default shake
