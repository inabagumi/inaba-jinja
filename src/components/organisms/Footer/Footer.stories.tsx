import React from 'react'
import Footer from './Footer'

export default {
  component: Footer,
  title: 'Footer'
}

export const footer = (): JSX.Element => <Footer />
