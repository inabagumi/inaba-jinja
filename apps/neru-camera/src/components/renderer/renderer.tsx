import { makeStyles } from '@material-ui/core/styles'
import { Application, Sprite } from 'pixi.js'
import { Viewport } from 'pixi-viewport'
import React, {
  ReactElement,
  SyntheticEvent,
  forwardRef,
  useCallback,
  useEffect,
  useImperativeHandle,
  useRef,
  useState
} from 'react'
import { Asset } from '../../context/asset-context'
import { ChromaKeyFilter } from '../../filters/ChromaKeyFilter'

const useStyles = makeStyles({
  container: {
    position: 'relative',
    width: '100%'
  },
  preloaded: {
    display: 'none'
  },
  preview: {
    display: 'block',
    height: 'auto',
    width: '100%'
  }
})

export type RefObject = {
  toBlob: (type: string, quality: number) => Promise<Blob>
}

export type Props = {
  asset: Asset
  cameraStream: MediaStream
}

const Renderer = forwardRef<RefObject, Props>(
  ({ asset, cameraStream }, ref): ReactElement => {
    const [viewport, setViewport] = useState<Viewport>()
    const classes = useStyles()

    const previewRef = useRef<HTMLCanvasElement>(null)
    const cameraRef = useRef<HTMLVideoElement>(null)
    const overlayRef = useRef<HTMLVideoElement>(null)

    const setup = useCallback(
      (event: SyntheticEvent<HTMLVideoElement, Event>) => {
        if (!previewRef.current) return

        const background = Sprite.from(event.currentTarget)

        const app = new Application({
          height: background.height,
          preserveDrawingBuffer: true,
          view: previewRef.current,
          width: background.width
        })
        const viewport = new Viewport({
          interaction: app.renderer.plugins.interaction,
          screenHeight: app.screen.height,
          screenWidth: app.screen.width,
          worldHeight: app.screen.height,
          worldWidth: app.screen.width
        })

        viewport
          .drag()
          .pinch()
          .wheel()
          .decelerate()

        app.stage.addChild(background)
        app.stage.addChild(viewport)

        setViewport(viewport)

        if (overlayRef.current) overlayRef.current.load()
      },
      []
    )

    const handleLoadedData = useCallback(
      (event: SyntheticEvent<HTMLVideoElement, Event>) => {
        if (!viewport) return

        const target = event.currentTarget
        const overlay = Sprite.from(target)
        const { keyColor } = target.dataset

        overlay.filters = [new ChromaKeyFilter(keyColor)]
        overlay.anchor.set(0.5)
        overlay.position.set(
          viewport.worldWidth * 0.5,
          viewport.worldHeight * 0.5
        )

        viewport.addChild(overlay)
      },
      [viewport]
    )

    useEffect(() => {
      if (!cameraRef.current) return

      cameraRef.current.srcObject = cameraStream
    }, [cameraStream])

    useImperativeHandle(ref, () => ({
      toBlob: (type = 'image/png', quality = 1.0): Promise<Blob> =>
        new Promise((resolve, reject): void => {
          if (!previewRef.current) {
            reject(new TypeError('Canvas is not mounted.'))
            return
          }

          const canvas = document.createElement('canvas')
          const context = canvas.getContext('2d')

          canvas.width = previewRef.current.width
          canvas.height = previewRef.current.height

          if (!context) {
            reject(new TypeError('Context does not exist.'))
            return
          }

          try {
            context.drawImage(previewRef.current, 0, 0)
            canvas.toBlob(
              blob => {
                if (!blob) {
                  reject(new TypeError('Blob is empty.'))
                  return
                }

                resolve(blob)
              },
              type,
              quality
            )
          } catch (error) {
            reject(error)
          }
        })
    }))

    return (
      <div className={classes.container}>
        <canvas className={classes.preview} ref={previewRef} />

        <video
          className={classes.preloaded}
          crossOrigin="anonymouse"
          muted
          onLoadedData={setup}
          playsInline
          ref={cameraRef}
        />
        <video
          className={classes.preloaded}
          crossOrigin="anonymouse"
          data-key-color={asset.keyColor}
          loop
          muted
          onLoadedData={handleLoadedData}
          playsInline
          preload="none"
          ref={overlayRef}
          src={asset.src}
        />
      </div>
    )
  }
)

export default Renderer
