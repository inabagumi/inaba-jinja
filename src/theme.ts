import deepOrange from '@material-ui/core/colors/deepOrange'
import orange from '@material-ui/core/colors/orange'
import { createMuiTheme } from '@material-ui/core/styles'

export default createMuiTheme({
  palette: {
    primary: deepOrange,
    secondary: orange,
    type: 'light'
  },
  typography: {
    body1: {
      lineHeight: 2
    },
    body2: {
      lineHeight: 2
    },
    fontFamily: [
      'Roboto',
      'Helvetica',
      'Arial',
      'Noto Sans JP',
      'sans-serif'
    ].join(',')
  }
})
