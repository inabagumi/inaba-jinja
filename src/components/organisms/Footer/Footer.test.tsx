/* eslint-env jest */

import { ThemeProvider } from '@material-ui/core/styles'
import React, { FC } from 'react'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import theme from '../../../theme'
import Footer from './Footer'

const App: FC = ({ children }) => (
  <IntlProvider locale="en">
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </IntlProvider>
)

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <App>
          <Footer />
        </App>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
