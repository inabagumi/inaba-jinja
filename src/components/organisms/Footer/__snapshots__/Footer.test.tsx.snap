// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
Array [
  <hr
    className="MuiDivider-root"
  />,
  <footer
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root makeStyles-logoGrid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <SvgMock
          aria-label="因幡神社"
          className="makeStyles-root-121 makeStyles-logo-1"
          focusable="false"
          role="img"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <nav
          className="makeStyles-navLinks-3"
        >
          <ul>
            <li>
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorTextSecondary"
                href="https://haneru.dev/"
                onBlur={[Function]}
                onFocus={[Function]}
                rel="noopener noreferrer"
                target="_blank"
              >
                Haneru Developers
              </a>
            </li>
            <li>
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorTextSecondary"
                href="/privacy"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
              >
                プライバシーポリシー
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </footer>,
]
`;
